crates/oxc_linter/src/rules/typescript/no_var_requires.rs

```diff
@@ -0,0 +1,107 @@
+use oxc_ast::AstKind;
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error(
+    "typescript-eslint(no-var-requires): Require statement not part of import statement."
+)]
+#[diagnostic(severity(error))]
+struct NoVarRequiresDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoVarRequires;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// 
+    /// Disallow `require` statements except in import statements
+    /// 
+    /// ### Why is this bad?
+    /// 
+    /// In other words, the use of forms such as var foo = require("foo") are banned. Instead use ES6 style imports or import foo = require("foo") imports.
+    /// 
+    /// ```javascript
+    /// var foo = require('foo');
+    /// const foo = require('foo');
+    /// let foo = require('foo');
+    /// ```
+    NoVarRequires,
+    correctness
+);
+
+impl Rule for NoVarRequires {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
```

Comment:
It seems like we need to check whether the source is typescript, otherwise all js files will trigger the warning.

Reply:
I have modified it to check the source type and return early, is this as intended?